### PR TITLE
Improve Attachment and ClassifiedTaxCategory

### PR DESCRIPTION
**Pull request by bagulho**
Allow Attachment to define the content of the file
Add missing accessors for ClassifiedTaxCategory

### CODE REVIEW

**Comment by kouinkouin** [comment]
Hi!
For the new property `Attachement::$fileContent`, do you think that could be better to accept a raw file content, instead of an base64-encoded string?
The name `fileContent` does not suggest the string must be base64-encoded.
What do you think?

File: src/Attachment.php [hunk 1]
```diff
@@ -89,9 +150,14 @@ public function xmlSerialize(Writer $writer): void
     {
         $this->validate();
 
-        if ($this->filePath) {
-            $fileContents = base64_encode(file_get_contents($this->filePath));
-            $mimeType = $this->getFileMimeType();
+        if ($this->filePath || $this->fileContent) {
+            if ($this->filePath) {
+                $fileContents = base64_encode(file_get_contents($this->filePath));
+                $mimeType = $this->getFileMimeType();
```
Comment (kouinkouin) [inline src/Attachment.php:156]: add:
```php
$filename = basename($this->filePath);
```

File: src/Attachment.php [hunk 2]
```diff
@@ -89,9 +150,14 @@ public function xmlSerialize(Writer $writer): void
...
+            } else {
+                $fileContents = $this->fileContent;
+                $mimeType = $this->mimeType;
```
Comment (kouinkouin) [inline src/Attachment.php:159]: add:
```php
$filename = $this->filename;
```

**Review by kouinkouin — CHANGES_REQUESTED** [review]
The property `$fileName` is not used in the class (excepted in get/set).
Maybe, in the method `XmlSerialize`, we could replace the line 167 with:
```diff
-                    'filename' => basename($this->filePath)
+                    'filename' => $filename
```
?

**Comment by kouinkouin** [comment]
Your PR interests me because that is absent of the main branch, so thanks for the PR :-)

**Comment by brtdv** [comment]
Hi @kouinkouin, thank you for your PR. 

I was going through pull requests today and I've merged something similar earlier today... I think #66 is basically this same functionality, correct? If yes, I'll close this one because I've merged this other one already and the fuctionality is very similar.

**Comment by Unubius** [comment]
@brtdv Hello, could you please merge the changes from `ClassifiedTaxCategory.php`, I would greatly appreciate it 🙏

**Comment by brtdv** [comment]
@Unubius, will check, I'm preparing release v1.18 right now and I'll include these

**Comment by brtdv** [comment]
Hi @kouinkouin, @bagulho, this PR had some overlap with #66. I kept the 'setFileStream' function on `Attachment` from that PR, which basically does the same as `setFileContents` proposed in this PR. This PR also included a setMimeType function which I kept. 

@Unubius all other changes from the PR, such as the changes in `ClassifiedTaxCategory` were included in this PR.

This all will be released in v1.18.

**Pull request merged**